src/main/java/org/zalando/nakadi/service/EventStream.java

```diff
@@ -109,6 +123,18 @@ public void streamEvents(final AtomicBoolean connectionReady, final Runnable che
                     }
                 }
 
+                if (lastKpiEventSent + kpiFrequencyMs < System.currentTimeMillis()) {
+                    final long count = kpiData.getAndResetNumberOfEventsSent();
+                    final long bytes = kpiData.getAndResetBytesSent();
+
+                    if (count > 0) {
```

Comment:
as I understand you skip the bytes which were sent by keep-alives (within empty batches), right?

Reply:
Yes, exactly. I thought that this is not actual event data and that it's mostly and operational overhead that the client has not much control over. What do you think? It wasn't specified but I think that this is what is expected.